ddtrace/internal/telemetry/writer.py

```diff
@@ -0,0 +1,221 @@
+        with StopWatch() as sw:
+            try:
+                conn.request("POST", ENDPOINT, rb_json, get_headers(request["request_type"]))
+                resp = get_connection_response(conn)
+
+                t = sw.elapsed()
+                if t >= self.interval:
+                    log_level = logging.WARNING
+                else:
+                    log_level = logging.DEBUG
+
+                log.log(
+                    log_level,
+                    "sent %d in %.5fs to %s/%s. response: %s",
+                    len(rb_json),
+                    t,
+                    AGENT_URL,
+                    ENDPOINT,
+                    resp.status,
+                )
+            except Exception:
+                log.error("failed to send telemetry to the Datadog Agent at %s", ENDPOINT, exc_info=True)
+            finally:
+                conn.close()
+        return resp
+
+    def flush_integrations_queue(self):
+        # type () -> List[Dict]
+        """Returns a list of all integrations queued by add_integration"""
+        with self._lock:
+            integrations = self._integrations_queue
+            self._integrations_queue = []
+        return integrations
+
+    def flush_events_queue(self):
+        # type () -> List[Dict]
+        """Returns a list of all integrations queued by classmethods"""
+        with self._lock:
+            requests = self._events_queue
+            self._events_queue = []
+        return requests
+
+    def periodic(self):
+        integrations = self.flush_integrations_queue()
+        if integrations:
+            self.app_integrations_changed_event(integrations)
+
+        telemetry_requests = self.flush_events_queue()
+
+        for telemetry_request in telemetry_requests:
+            self._send_request(telemetry_request)
+
+    def shutdown(self):
+        # type: () -> None
+        self.app_closed_event()
+        self.periodic()
+
+    @classmethod
+    def add_event(cls, payload, payload_type):
+        # type: (Dict, str) -> None
+        """
+        Adds a Telemetry Request to the TelemetryWriter request buffer
+
+        :param Dict: stores a formatted telemetry request
+        """
+        with cls._lock:
+            if cls._instance is None:
+                return
+
+            request = create_telemetry_request(payload, payload_type, cls.sequence)
+            cls.sequence += 1
+            cls._instance._events_queue.append(request)
+
+    @classmethod
+    def add_integration(cls, integration_name):
+        # type: (str) -> None
+        """
+        Creates and queues the names and settings of a patched module
+
+        :param str integration_name: name of patched module
+        """
+        with cls._lock:
+            if cls._instance is None:
+                return
+
+            integration = {
+                "name": integration_name,
+                "version": "",
+                "enabled": True,
+                "auto_enabled": True,
```

Comment:
This should only be `True` if the default patch value in `_monkey.py` is `True`.

https://github.com/DataDog/dd-trace-py/blob/13f9c6c1a8b4820365b299ab204f2bb5189d2a49/ddtrace/_monkey.py#L20-L76

We should update `add_integration` to take in `integration_name, auto_enabled` and pass in whether the module is `True` in the `PATCH_MODULES` dict.